core/beacon/src/types.rs

```diff
@@ -48,15 +48,27 @@ impl BeaconBlock {
                 parent_hash,
                 merkle_root_tx: MerkleHash::default(),
                 merkle_root_state,
-                signature: primitives::signature::default_signature(),
+                signature: DumbGroupSignature::default(),
                 index,
             },
             transactions,
         }
     }
 
-    pub fn sign(&mut self, signer: &Arc<Signer>) {
-        self.header.signature = signer.sign(&self.hash());
+    // Since the signature is contained in the header, we need a hash that omits it
+    fn hash_for_signing(&self) -> CryptoHash {
+        // TODO: find a better way to do this
+        let dummy_block = BeaconBlockHeader{signature: DumbGroupSignature::default(), ..self.header};
```

Comment:
we should change this, so don't need to hash default value. In #112 started chaging to Header and SignedHeader model. 
I'm also planning to add shard chains as data structure for transactions / merkle tree (e.g. creates together with Beacon blocks), then I think we don't need Header/Body split here.

Reply:
Was my "todo" not clear enough?  Obviously hashing the default signature is silly.  What should I do here to minimally conflict with you?